src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -81,10 +83,16 @@ GcsServer::GcsServer(const ray::gcs::GcsServerConfig &config,
     RAY_LOG(FATAL) << "Unexpected storage type: " << storage_type_;
   }
 
+  // Init KV Manager. This needs to be initialized first here so that
+  // it can be used to retrieve the cluster ID.
+  InitKVManager();
+  CacheAndSetClusterId();
```

Comment:
Also a high level question for the order of things:

1. Init kv_manager
2. get or set cluster id
3. create rpc server with the known cluster id

Would this be cleaner since rpc server is always created with a cluster id instead of setting it afterwords with `SafeClusterID`

Reply:
Ideally, yes. But the RPC server has to exist to init KV manager.

Reply:
Ah. You are right, it is very simple to separate out

Reply:
Several things:
1. Right now, Object Manager and Node Manager use GRPC server without Cluster ID. They may not get Cluster ID until after GRPC server is initialized, so the SetClusterID model instead of the pass-during-initialization model makes more sense to me.
2. I added a check to make sure the Cluster ID is set synchronously.